src/dotnet-interactive-vscode/src/stdioKernelTransport.ts

```diff
@@ -58,29 +58,36 @@ export class StdioKernelTransport {
                 }
             }
         });
+    }
 
-        // set the working directory to be next to the notebook; this allows relative file access to work as expected
-        // immediately clean up afterwards because we'll never do this again
-        let token = 'change-working-directory-token';
-        let disposable = this.subscribeToKernelEvents(envelope => {
-            if (envelope.command?.token === token) {
-                switch (envelope.eventType) {
-                    case CommandFailedType:
-                        let failed = <CommandFailed>envelope.event;
-                        let message = `Unable to set notebook working directory to '${notebookPath}'.\n${failed.message}`;
-                        diagnosticChannel.appendLine(message);
-                        disposable.dispose();
-                        break;
-                    case CommandHandledType:
-                        disposable.dispose();
-                        break;
+    static create(processStart: ProcessStart, notebookPath: string, diagnosticChannel: ReportChannel): Promise<StdioKernelTransport> {
+        return new Promise<StdioKernelTransport>((resolve, reject) => {
+            let kernelTransport = new StdioKernelTransport(processStart, notebookPath, diagnosticChannel);
+            // set the working directory to be next to the notebook; this allows relative file access to work as expected
+            // immediately clean up afterwards because we'll never do this again
+            let token = 'change-working-directory-token';
+            let disposable = kernelTransport.subscribeToKernelEvents(envelope => {
+                if (envelope.command?.token === token) {
+                    switch (envelope.eventType) {
+                        case CommandFailedType:
+                            let failed = <CommandFailed>envelope.event;
+                            let message = `Unable to set notebook working directory to '${notebookPath}'.\n${failed.message}`;
+                            diagnosticChannel.appendLine(message);
+                            reject();
```

Comment:
In this scenario, the initial CWD command failed.  Should we still call `resolve()` and pretend it was OK, or do we fail and not even open the notebook?

Reply:
Nevermind; I tried this scenario where CWD fails and with the `reject()` call the notebook immediately closes with no actionable message to the user.  That doesn't seem very useful so I changed this to `resolve(...)` so the notebook can still be opened, and the failure message at least appears in the diagnostic log below.